File: app/src/main/java/com/example/ls/maillistdemo/maillist/StickyHeaderAdapter.java
package com.example.ls.maillistdemo.maillist;

import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

/**
 * Created by LS on 2017/12/29.
 */

public interface StickyHeaderAdapter<T extends RecyclerView.ViewHolder> {


    String getHeaderId(int position);


    T onCreateHeaderViewHolder(ViewGroup parent);


    void onBindHeaderViewHolder(T viewholder, int position);
}

File: app/src/main/java/com/example/ls/maillistdemo/maillist/StickyHeaderAdapter.kt
package com.example.ls.maillistdemo.maillist

import android.support.v7.widget.RecyclerView
import android.view.ViewGroup

/**
 * Created by LS on 2017/12/29.
 */

interface StickyHeaderAdapter<T : RecyclerView.ViewHolder> {


    fun getHeaderId(position: Int): String?


    fun onCreateHeaderViewHolder(parent: ViewGroup): T


    fun onBindHeaderViewHolder(viewholder: T, position: Int)
}
